src/storage_utils.cpp

```diff
@@ -63,19 +63,20 @@ namespace libtorrent { namespace aux {
 		return ret;
 	}
 
-	typed_span<iovec_t> advance_bufs(typed_span<iovec_t> bufs, int const bytes)
+	typed_span<iovec_t> advance_bufs(typed_span<iovec_t> const& bufs, int const bytes)
 	{
```

Comment:
when stepping past this line "{", ``&bufs`` changed from being a reasonable pointer to being a pointer to 0xa (which happens to be the value of the``m_len`` field). At first I thought it was some subtle conversion issue, where it ended up copying a temporary that had gone out of scope. But I could not come up with an explanation for that. Stepping through the debugger indicated that it was all good at the time of entering this function.